source/stun_serializer.c

```diff
@@ -907,20 +928,30 @@ StunResult_t StunSerializer_Finalize( StunContext_t * pCtx,
                                       size_t * pStunMessageLength )
 {
     StunResult_t result = STUN_RESULT_OK;
+    size_t messageLength;
```

Comment:
On some platforms size_t can be 16-bit. Thus, will it be able to catch the `if( messageLength > UINT16_MAX )`? Will it not overflow?